syntax = "proto3";

option go_package = "github.com/easyp-tech/course-grpc/pkg/api/stream/v1";

package api.stream.v1;

message EchoRequest {
  string message = 1;
};

message EchoResponse {
  string message = 1;
};

service EchoService {
  rpc EchoClientStream(stream EchoRequest) returns (EchoResponse);
  rpc EchoServerStream(EchoRequest) returns (stream EchoResponse);
  rpc EchoBidirectionalStreamSync(stream EchoRequest) returns (stream EchoResponse);
  rpc EchoBidirectionalStreamAsync(stream EchoRequest) returns (stream EchoResponse);
}
